### PR TITLE
2.4 cherry pick for azure adls plugins

### PR DESCRIPTION
**Pull request by shifuxu0301**
cherry pick azure adls changes into 2.4 branch. Changes mainly includes add azure as submodule and update the release branch from 1.6 to 1.7.  Update submodule is not included.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -31,6 +31,10 @@
 	path = amazon-s3-plugins
 	url = ../../data-integrations/amazon-s3-plugins
 	branch = release/1.12
+[submodule "azure"]
+	path = azure
+	url = ../../data-integrations/azure
+	branch = release/1.7
```
Comment (elfenheart) [inline .gitmodules]: @shifuxu0301 can you change this to release/1.8 and update the submodule hash?

Comment (shifuxu0301) [reply .gitmodules]: My ongoing PR is not merged into release/1.8 yet. And also the version is also not bumped yet. In this particular case, is it okay for me to upgrade to 1.8?

Comment (elfenheart) [reply .gitmodules]: yeah, that's ok. We want the azure shipped with 6.2.0 to be version 1.8

Comment (shifuxu0301) [reply .gitmodules]: Got it. Will do.

**Review by elfenheart — APPROVED** [review]

**Comment by shifuxu0301** [comment]
close this one

**Pull request closed without merging**